.github/workflows/branch-sync.yml

```diff
@@ -13,71 +13,90 @@ permissions:
+          ( echo ; echo "EOF" ) | tee -a "$GITHUB_OUTPUT"
 
-          # Merge 6.x into temp branch, preferring 7.x on conflicts
-          if ! git merge --no-ff -X ours origin/6.x -m "Sync 6.x into 7.x (prefer 7.x on conflicts)"; then
-            echo "Merge reported conflicts; force-resolving all to 7.x (ours)..."
-            HAD_CONFLICTS=true
-            # Accept 'ours' version for all remaining conflicts
-            git checkout --ours .
-            git add -A
-            git commit -m "Sync 6.x into 7.x (auto-resolved conflicts preferring 7.x)"
+          if [[ $diff == "false" ]]; then
+            echo "There are no commits between the base and HEAD branches." >> $GITHUB_STEP_SUMMARY
+            COMMAND_STATUS=0
           fi
 
-          # If the merge results in no net change, don't open a PR
-          if git diff --quiet origin/7.x..HEAD; then
-            echo "NO_CHANGES=true" >> "${GITHUB_ENV}"
-          fi
+          echo "diff=$diff" >> "$GITHUB_OUTPUT"
 
-          if $HAD_CONFLICTS; then
-            echo "HAD_CONFLICTS=true" >> "${GITHUB_ENV}"
-          fi
-
-      - name: Create a pull request from sync branch to 7.x
-        if: env.NO_CHANGES != 'true'
-        uses: peter-evans/create-pull-request@v7
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          base: 7.x
-          branch: temp-sync-branch
-          title: "Sync changes from 6.x"
-          body: |
-            This pull request syncs relevant changes from the `6.x` branch into the `7.x` branch.
-
-            - Merge performed with `-X ours` to prefer `7.x` on conflicts.
-            - Any remaining conflicts were auto-resolved to `ours` to keep CI green; reviewers can adjust in this PR if needed.
-            - Non-conflicting changes from `6.x` are preserved.
+          exit $COMMAND_STATUS
+        shell: bash
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-            This PR was automatically generated by the CI workflow.
-          labels: ${{ env.HAD_CONFLICTS == 'true' && 'sync, automated, merge-conflicts-resolved' || 'sync, automated' }}
-          draft: false
-          sign-commits: true
+      - name: If successful set PR number to env var
+        if: steps.create-pr.conclusion == 'success' && steps.create-pr.outputs.diff == 'true'
+        run: |
+          output="${{ steps.create-pr.outputs.stdout }}"
+          pr_url=$(echo $output | grep -o "^https.*[0-9]")
+          pr=(echo $pr_url | cut -d "/" -f 7)
+          
+          echo "PR_URL=$pr_url" >> GITHUB_ENV
+          echo "PR_NUMBER=$pr" >> $GITHUB_ENV
 
-      - name: Enable auto-merge
-        if: env.NO_CHANGES != 'true'
+      - name: Successful, check for conflicts
+        if: env.PR_NUMBER
         run: |
-          gh pr merge temp-sync-branch --auto --merge
+          conflicts=$(gh pr view ${{ env.PR_NUMBER }} --json mergeStateStatus --jq 'if .mergeStateStatus == "DIRTY" then true else false end')
```

Comment:
This command returns the latest PR for the currently checked out branch, 6.x in this case.